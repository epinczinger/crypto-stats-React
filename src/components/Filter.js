import React from 'react';
import PropTypes from 'prop-types';

const Filter = ({ handleFilter }) => (
  <div className="mx-auto pt-2 mb-2 text-center">
    <label htmlFor="filter" name="filter" className="text-xl">
      Filter Coins:
      <select
        name="filter"
        className="dark:bg-gray-700 ml-2 p-2 bg-gray-200 rounded"
        onChange={handleFilter}
      >
        <option value="All" className="text-xl text-gray-700">
          All
        </option>
        <option value="Bullish" className="text-xl text-green-600">
          Bullish
        </option>
        <option value="Bearish" className="text-xl text-red-700">
          Bearish
        </option>
      </select>
    </label>
  </div>
);

export default Filter;

Filter.propTypes = {
  handleFilter: PropTypes.func,
};

Filter.defaultProps = {
  handleFilter: () => {},
};
