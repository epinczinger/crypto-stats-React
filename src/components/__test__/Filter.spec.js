import React from "react";
import renderer from "react-test-renderer";
import Filter from "../Filter";

describe("The Filter", () => {
  it("renders as expected matching snapshot", () => {
    const tree = renderer.create(<Filter handleFilter={Function}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
