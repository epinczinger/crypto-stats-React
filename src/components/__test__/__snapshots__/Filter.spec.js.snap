// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Filter renders as expected matching snapshot 1`] = `
<div
  className="mx-auto pt-2 mb-2 text-center"
>
  <label
    className="text-xl"
    htmlFor="filter"
    name="filter"
  >
    Filter Coins:
    <select
      className="dark:bg-gray-700 ml-2 p-2 bg-gray-200 rounded"
      name="filter"
      onChange={[Function]}
    >
      <option
        className="text-xl text-gray-700"
        value="All"
      >
        All
      </option>
      <option
        className="text-xl text-green-600"
        value="Bullish"
      >
        Bullish
      </option>
      <option
        className="text-xl text-red-700"
        value="Bearish"
      >
        Bearish
      </option>
    </select>
  </label>
</div>
`;
